packages/filebrowser/test/listing.spec.ts

```diff
@@ -0,0 +1,284 @@
+        itemNodes[0].getBoundingClientRect = (): any => ({
+          left: 0,
+          right: 10,
+          top: 0,
+          bottom: 10
+        });
+        itemNodes[1].getBoundingClientRect = (): any => ({
+          left: 0,
+          right: 10,
+          top: 10,
+          bottom: 20
+        });
+        const checkboxes = itemNodes.map(node =>
+          dirListing.renderer.getCheckboxNode!(node)
+        ) as HTMLInputElement[];
+        const items = toArray(dirListing.sortedItems());
+        expect(dirListing.isSelected(items[0].name)).toBe(false);
+        expect(dirListing.isSelected(items[1].name)).toBe(false);
+        simulate(checkboxes[0], 'mousedown', {
+          clientX: 1,
+          clientY: 1
+        });
+        simulate(checkboxes[1], 'mousedown', {
+          clientX: 1,
+          clientY: 11
+        });
+        await signalToPromise(dirListing.updated);
+        expect(dirListing.isSelected(items[0].name)).toBe(true);
+        expect(dirListing.isSelected(items[1].name)).toBe(true);
+      });
+
+      it('should reflect multiple items selected', async () => {
+        const itemNodes = Array.from(
+          dirListing.contentNode.children
+        ) as HTMLElement[];
+        const checkboxes = itemNodes.map(node =>
+          dirListing.renderer.getCheckboxNode!(node)
+        ) as HTMLInputElement[];
+        expect(checkboxes[0].checked).toBe(false);
+        expect(checkboxes[1].checked).toBe(false);
+        dirListing.selectNext();
+        dirListing.selectNext(true); // true = keep existing selection
+        await signalToPromise(dirListing.updated);
+        expect(checkboxes[0].checked).toBe(true);
+        expect(checkboxes[1].checked).toBe(true);
+      });
+
+      // A double click on the item should open the item; however, a double
+      // click on the checkbox should only check/uncheck the box.
+      it('should not open item on double click', () => {
+        const itemNode = dirListing.contentNode.children[0] as HTMLElement;
+        const checkbox = dirListing.renderer.getCheckboxNode!(
+          itemNode
+        ) as HTMLInputElement;
+        const wasOpened = jest.fn();
+        dirListing.onItemOpened.connect(wasOpened);
+        simulate(checkbox, 'dblclick');
+        expect(wasOpened).not.toHaveBeenCalled();
+        dirListing.onItemOpened.disconnect(wasOpened);
+      });
+
+      it('should not become unchecked due to right-click on selected item', async () => {
+        const itemNode = dirListing.contentNode.children[0] as HTMLElement;
+        itemNode.getBoundingClientRect = (): any => ({
+          left: 0,
+          right: 10,
+          top: 0,
+          bottom: 10
+        });
+        const checkbox = dirListing.renderer.getCheckboxNode!(
+          itemNode
+        ) as HTMLInputElement;
+        const item = dirListing.sortedItems().next()!;
+        dirListing.selectItemByName(item.name);
```

Comment:
Missing `await` here?

```suggestion
        await dirListing.selectItemByName(item.name);
```